share/smack/lib/fenv.c

```diff
@@ -0,0 +1,26 @@
+#include <fenv.h>
+#include <smack.h>
+
+int fegetround()
+{
+  int ret = __VERIFIER_nondet_int();
+  __SMACK_code("if ($rmode == RNE) {@ := 1;}", ret);
+  __SMACK_code("if ($rmode == RNA) {@ := 2;}", ret);
+  __SMACK_code("if ($rmode == RTP) {@ := 3;}", ret);
+  __SMACK_code("if ($rmode == RTN) {@ := 4;}", ret);
+  __SMACK_code("if ($rmode == RTZ) {@ := 5;}", ret);
+  return ret;
+}
+
+int fesetround(int rm)
+{
+  switch (rm) {
```

Comment:
Here is what the spec I found says:
```
Changes the currently selected rounding mode to round. If round does not correspond to one of the supported rounding modes nothing is changed. fesetround returns zero if it changed the rounding mode, or a nonzero value if the mode is not supported.
```
So this code should be changed to return a non-zero value if the mode is not supported (in which case it also does not get changed). I hope this makes sense.